### PR TITLE
Added doc for bipartite check

### PR DESCRIPTION
**Pull request by Jahnavi-Majji**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added documentation for the Bipartite Check problem,

## Related Issues or Pull Requests

Fixes issue #33

### CODE REVIEW

File: docs/graph-algorithms/bipartite-check.mdx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+## Introduction
+
+A bipartite graph is a graph whose vertices can be divided into two disjoint sets so that every edge connects two vertices from different sets (i.e. there are no edges which connect vertices from the same set). These sets are usually called sides.
+
+A graph is bipartite if and only if it is two-colorable. Equivalently, a bipartite graph is a graph that does not contain any *odd-length cycles*.
+
+In the Bipartite check problem, we are given an undirected graph. We need to determine whether the given graph is a Bipartite.
+
+[![Screenshot-82.png](https://i.postimg.cc/W4LJJ80T/Screenshot-82.png)](https://postimg.cc/QFmM2p6y)
```
Comment (HarshCasper) [inline docs/graph-algorithms/bipartite-check.mdx:14]: Have you made this image?

File: docs/graph-algorithms/bipartite-check.mdx [hunk 2]
```diff
@@ -0,0 +1,245 @@
+---
+title: Bipartite Check
+slug: /graphs-algorithms/bipartite-check
+---
+
+## Introduction
+
+A bipartite graph is a graph whose vertices can be divided into two disjoint sets so that every edge connects two vertices from different sets (i.e. there are no edges which connect vertices from the same set). These sets are usually called sides.
+
+A graph is bipartite if and only if it is two-colorable. Equivalently, a bipartite graph is a graph that does not contain any *odd-length cycles*.
```
Comment (HarshCasper) [inline docs/graph-algorithms/bipartite-check.mdx:10]: Keep `odd-length cycles` in the bold

File: docs/graph-algorithms/bipartite-check.mdx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+
+
+## Algorithm
+
+We can use a series of BFS to determine if the given graph is Bipartite
+- Create an empty array `color` which indicates the color of the vertex.
+- Initialize the `color` array to `-1` which means at the start all vertices are uncolored.
+- From every uncolored vertex 
+    - Perform BFS. Assign the vertex the color `1` (putting into one set `U`).
```
Comment (HarshCasper) [inline docs/graph-algorithms/bipartite-check.mdx:26]: `Perform Breadth First Search`

File: docs/graph-algorithms/bipartite-check.mdx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+
+## Algorithm
+
+We can use a series of BFS to determine if the given graph is Bipartite
+- Create an empty array `color` which indicates the color of the vertex.
+- Initialize the `color` array to `-1` which means at the start all vertices are uncolored.
+- From every uncolored vertex 
+    - Perform BFS. Assign the vertex the color `1` (putting into one set `U`). 
+    - Color its neighbors with color `color ^ 1` (putting into the other set `V`).
```
Comment (HarshCasper) [inline docs/graph-algorithms/bipartite-check.mdx:27]: This doesn't look understandable. Can you reword?

File: docs/graph-algorithms/bipartite-check.mdx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+```
+Enter number of vertices and edges: 4 4
+Enter the source and destination of the edges: 
+1 2
+2 3
+4 3
+1 4
+```
+
+### Sample Output
```
Comment (HarshCasper) [inline docs/graph-algorithms/bipartite-check.mdx:229]: Keep Heading-4

File: docs/graph-algorithms/bipartite-check.mdx [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+    print("Is the graph Bipartite? ", ans)
+
+```
+
+</TabItem>
+</Tabs>
+
+## Sample I/O
+
+### Sample Input
```
Comment (HarshCasper) [inline docs/graph-algorithms/bipartite-check.mdx:218]: Keep Heading-4

File: docs/graph-algorithms/bipartite-check.mdx [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+    # Print the Answer
+    print("Is the graph Bipartite? ", ans)
+
+```
+
+</TabItem>
+</Tabs>
+
+## Sample I/O
```
Comment (HarshCasper) [inline docs/graph-algorithms/bipartite-check.mdx:216]: Keep Sample Input and Output

File: docs/graph-algorithms/bipartite-check.mdx [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+        # For each uncolored node, we have to call BFS() function
+        if color[i] == -1:
+
+            # Answer will be the (and) of the previous and the boolean value returned
+            # by the BFS() function
+            answer = answer and BFS(graph, i, color)
+
+    return answer
+
+# -----------------------------------Driver Code--------------------------
```
Comment (HarshCasper) [inline docs/graph-algorithms/bipartite-check.mdx:194]: Remove this

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/graph-algorithms/bipartite-check.mdx (hunk 1 above)
Comment (Jahnavi-Majji) [reply docs/graph-algorithms/bipartite-check.mdx:14]: Yes. I used [this](https://postimg.cc/) to get the url.

**Comment by Jahnavi-Majji** [comment]
@HarshCasper  requested changes are made in the documentation.

**Comment by Jahnavi-Majji** [comment]
@HarshCasper the build is failing on my local machine and I'm closing this PR and will open a new PR.

**Pull request closed without merging**